### PR TITLE
docs: add examples for NewStoreFromDocker and NewStoreWithFallbacks

### LINKED ISSUE #49: Create examples
Examples are needed for:

- [ ] `NewNativeStore`
- [ ] `NewFileStore`
- [ ] `NewStore`
- [ ] `NewStoreFromDocker`
- [ ]  `NewStoreWithFallbacks`
- [ ] `Login`
- [ ] `Logout`
- [ ] `Credential`

### PR DESCRIPTION
**Pull request by wangxiaoxuan273**
Resolves #49

### CODE REVIEW

File: example_test.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	"fmt"
+
+	credentials "github.com/oras-project/oras-credentials-go"
+	"oras.land/oras-go/v2/registry/remote/auth"
+)
+
+func ExampleNewStoreFromDocker() {
+	// NewStoreFromDocker returns a Store based on the default docker config file.
+	// See the documentation for details.
+	ds, err := credentials.NewStoreFromDocker(credentials.StoreOptions{AllowPlaintextPut: true})
```
Comment (Wwwsylvia) [inline example_test.go:27]: Maybe we don't need these comments here.
image

File: example_test.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+	if err != nil {
+		panic(err)
+	}
+}
+
+func ExampleNewStoreWithFallbacks() {
+	// NewStoreWithFallbacks returns a new store based on the given stores.
+	// The first store is used as the primary store, and the rest as fallbacks.
+	// See the documentation for details.
+	primaryStore, err := credentials.NewStore("example/path/config.json", credentials.StoreOptions{AllowPlaintextPut: true})
```
Comment (Wwwsylvia) [inline example_test.go:59]: Same here.

File: example_test.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	if err != nil {
+		panic(err)
+	}
+	fallbackStore, err := credentials.NewStoreFromDocker(credentials.StoreOptions{AllowPlaintextPut: true})
```
Comment (Wwwsylvia) [inline example_test.go:63]: I think we can leave the StoreOptions empty. Since the credentials will never be put in the fallback store.

**Review by Wwwsylvia — CHANGES_REQUESTED** [review]

File: example_test.go (hunk 2 above)
Comment (wangxiaoxuan273) [reply example_test.go:59]: deleted

File: example_test.go (hunk 1 above)
Comment (wangxiaoxuan273) [reply example_test.go:27]: deleted

**Review by Wwwsylvia — APPROVED** [review]
LGTM

File: example_test.go [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+import (
+	"context"
+	"fmt"
+
+	credentials "github.com/oras-project/oras-credentials-go"
+	"oras.land/oras-go/v2/registry/remote/auth"
+)
+
+func ExampleNewStoreFromDocker() {
+	ds, err := credentials.NewStoreFromDocker(credentials.StoreOptions{AllowPlaintextPut: true})
```
Comment (shizhMSFT) [inline example_test.go:25]: Let's format the code in standard go style.
```suggestion
	ds, err := credentials.NewStoreFromDocker(credentials.StoreOptions{
		AllowPlaintextPut: true,
	})
```

File: example_test.go [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+	if err != nil {
+		panic(err)
+	}
+
+	// save credentials into the store
+	err = ds.Put(context.Background(), "localhost:8080",
+		auth.Credential{
+			Username: "username-example",
+			Password: "password-example"})
```
Comment (shizhMSFT) [inline example_test.go:34]:
```suggestion
	err = ds.Put(context.Background(), "localhost:8080", auth.Credential{
		Username: "username-example",
		Password: "password-example",
	})
```

File: example_test.go (hunk 4 above)
Comment (shizhMSFT) [reply example_test.go:25]: same for the other example

File: example_test.go (hunk 5 above)
Comment (shizhMSFT) [reply example_test.go:34]: same for the other example

File: example_test.go [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+		panic(err)
+	}
+	fmt.Println(cred)
+
+	// delete the credentials from the store
+	err = ds.Delete(context.Background(), "localhost:8080")
+	if err != nil {
+		panic(err)
+	}
+}
```
Comment (shizhMSFT) [inline example_test.go:51]: The output is not checked.

You need a `// Ouptut:` block.

File: example_test.go [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+	if err != nil {
+		panic(err)
+	}
+
+	// get credentials from the store
+	cred, err := sf.Get(context.Background(), "localhost:8080")
+	if err != nil {
+		panic(err)
+	}
+	fmt.Println(cred)
```
Comment (shizhMSFT) [inline example_test.go:75]: Getting from fallback store is not demonstrated.

Comment (wangxiaoxuan273) [reply example_test.go:75]: Changed the function name to `ExampleNewStoreWithFallbacks_configAsPrimaryStoreDockerAsFallback` to demonstrate the scenario.

File: example_test.go (hunk 6 above)
Comment (wangxiaoxuan273) [reply example_test.go:51]: Intentionally removed `// Output` so that the examples are not run as unit tests.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**